### PR TITLE
Andlrutt/tkw 61 filter trees table

### PR DESCRIPTION
**Pull request by andlrutt**
This is an extension of TKW-58, filtering trees by species name. Since this just takes that code and adds directly to it, I've closed the PR I had up for it since it would have immediately been overwritten by this PR. [Here](https://github.com/hack4impact-utk/trees-knoxville/pull/27) is the PR for TKW-58 if you would like to reference it.

This PR fulfills TKW-61, filtering the tree table based on age and date (on top of filtering by species name, which is what the above pull request is for).

Accomplishing this was difficult, there were a lot of things that made it more complicated than it would seem: An age/date filter was conditionally applied, but a species name filter was always applied, `stateinterface` values aren't immediately available for use after updating them, there were a lot of edge cases if an input field was empty, etc. A lot of things to work out. I've done my best to comment the code well, please let me know if there is anything I can clarify in terms of what is going on or why I did something the way I did.

Basically, whenever an input value is changed, a new filter is applied. The search bar for species is always displayed/applied, but there is a dropdown to select if you also want to filter by age or date. Take a look below to see exactly what's going on. In the video, the ages of the three trees displayed from top to bottom are 5, 10, and 15, respectively (also labeled in their location for convenience), and their date planted is visible. I go through all use cases to show how it works.

https://user-images.githubusercontent.com/71574118/163264703-a04ba54d-eb4a-4166-9c2d-6ed3346164bf.mp4

I held off on doing the frontend for this, pending a redesign of the whole tree table page.

### CODE REVIEW

**Review by tbatk — APPROVED** [review]
This is awesome! Really great work. Thank you for getting this done.
